### PR TITLE
Allow a field to represent an array of values

### LINKED ISSUE #60: allow a field to represent an array of values
With the addition of `checkbox-group` - https://github.com/UKHomeOffice/passports-template-mixins/pull/73 a field will be able to represent an array of values.  This should be taken into account when formatting values.

A change will also need to be made to the 'equal' validator which checks a value is present in the list of options - we will now need to check each value appears in the list.

IMO no changes need to be made to they way validators work as the values will be predefined in field config, the only relevant validators will be `required` which will work out of the box, and potentially a new validator which checks the number of values (select 2 etc). However this can be implemented as a custom validator as unlikely to be a common use case

### PR DESCRIPTION
**Pull request by joefitter**
This PR allows a field to represent an array of values. Formatters are applied to each value, the equal validator now checks each value is present in the options array

* convert value to array and apply formatters to each item
* updated the 'equal' validator to check each value if array
* added tests

### CODE REVIEW

**Comment by joefitter** [comment]
@JoeChapman @easternbloc @gavboulton @HughePaul

**Comment by gavboulton** [comment]
Looking.

**Comment by JoeChapman** [comment]
looking

File: test/spec/spec.form.js [hunk 1]
```diff
@@ -324,6 +324,18 @@ describe('Form Controller', function () {
             fn.should.not.throw();
         });
 
+        it('applies formtter to array of values', function () {
```
Comment (gavboulton) [inline test/spec/spec.form.js]: Typo 'formtter' to 'formatter'.

**Comment by gavboulton** [comment]
@joefitter,
@oli-weighell2 has done some work on multiple checkboxes so I'd like for him to have a look at this too. He's away at the moment but will be back next week.

**Comment by joefitter** [comment]
@gavboulton sure, thanks. It would be good to get more opinions on validating an array of values - my initial thoughts for checkboxes is they are predefined in field config so the only validation really required would be `required`. We could add another validator to check the amount of options checked, or this could be added as a custom validator in field config.

If there are any other use cases where multiple values are to be associated with one field key in the future, specifically raw input, then we would need to expand the validation logic to loop over values. But IMO I cant see a use case for this as this time

**Comment by joefitter** [comment]
@gavboulton is @oli-weighell2 back and able to look at this? The linked PR is https://github.com/UKHomeOffice/passports-template-mixins/pull/73

**Comment by gavboulton** [comment]
He's ill unfortunately but hopefully back within the next couple of days.

**Comment by oli-weighell2** [comment]
Looking

**Comment by oli-weighell2** [comment]
Looks good, covers our needs nicely - cheers! 
Couple of comments in the template-mixins one.

**Comment by easternbloc** [comment]
LGTM 👍 @gavboulton @oli-weighell2 can I go ahead and merge this I need to get a release of HOF out and we need this for it. 💌

**Comment by JoeChapman** [comment]
Can I merge it?

**Comment by easternbloc** [comment]
@JoeChapman I just want a passports OK on the last commit then I can't see why not...

File: lib/validation/index.js [hunk 2]
```diff
@@ -72,7 +72,12 @@ function validator(fields) {
                     value: true
                 };
             }
-            if (!dependent || (dependent && !fields[dependent.field]) || (dependent && values[dependent.field] === dependent.value)) {
+            if (!dependent
+                || (dependent && !fields[dependent.field])
+                || (dependent && (Array.isArray(values[dependent.field])
+                    ? values[dependent.field].indexOf(dependent.value) > -1
+                    : values[dependent.field] === dependent.value))
```
Comment (gavboulton) [inline lib/validation/index.js]: Should there be a test for this?

Comment (joefitter) [inline lib/validation/index.js]: there is now!

**Comment by joefitter** [comment]
@gavboulton test added

**Comment by easternbloc** [comment]
LGTM @gavboulton ?

File: test/spec/spec.form.js [hunk 3]
```diff
@@ -1079,6 +1091,54 @@ describe('Form Controller', function () {
                 cb.should.have.been.calledWith();
             });
 
+            describe('If a field represents an array of values', function () {
```
Comment (gavboulton) [inline test/spec/spec.form.js]: Would 'fields that are an array of values' be better?

Comment (joefitter) [inline test/spec/spec.form.js]: updated

**Comment by joefitter** [comment]
@gavboulton updated

File: test/spec/spec.form.js [hunk 4]
```diff
@@ -1079,6 +1091,54 @@ describe('Form Controller', function () {
                 cb.should.have.been.calledWith();
             });
 
+            describe('Fields that are an array of values', function () {
```
Comment (gavboulton) [inline test/spec/spec.form.js]: fields

**Comment by joefitter** [comment]
@gavboulton updated

**Comment by joefitter** [comment]
@gavboulton could you please publish a new tag for this change as we want to include it in hof release?

**Comment by gavboulton** [comment]
hmpo-form-controller@0.12.0 published.

**Pull request merged**